Tasks/Common/SonarQubeHelpers/SonarQubeHelper.ps1

```diff
@@ -12,22 +12,30 @@ function EscapeArg
 
 
 # Set a variable in a property bag that is accessible by all steps
-# To retrieve the variable use $val = Get-Variable $distributedTaskContext "varName"
+# To retrieve the variable use GetTaskContextVariable
+# The variable will be available in both the current and the subsequent tasks
 function SetTaskContextVariable
 {
     param([string][ValidateNotNullOrEmpty()]$varName, 
           [string]$varValue)
         
-    #[Environment]::SetEnvironmentVariable($varName, $varValue)
-    Write-Host "##vso[task.setvariable variable=$varName;]$varValue"
+    [Environment]::SetEnvironmentVariable($varName, $varValue)
+    Set-VstsTaskVariable $varName $varValue
 }
 
 function GetTaskContextVariable()
 {
 	param([string][ValidateNotNullOrEmpty()]$varName)
         
-    #return ([Environment]::GetEnvironmentVariable($varName))
-	return Get-TaskVariable -Context $distributedTaskContext -Name $varName    
+    $value = [Environment]::GetEnvironmentVariable($varName);
+
+    if ([String]::IsNullOrWhiteSpace($value))
+    {
+        $value = Get-VstsTaskVariable -Name $varName 
+    } 
+
+    Write-Verbose "Variable read: $varName = $value"
```

Comment:
Note, the lib function provides verbose tracing already. And also exposes additional switches (e.g. `-Require`) that you may find useful.